packages/react-core/src/components/NumberInput/__tests__/NumberInput.test.tsx

```diff
@@ -60,4 +61,82 @@ describe('numberInput', () => {
     );
     expect(asFragment()).toMatchSnapshot();
   });
+
+  test('renders negative value', () => {
+    render(<NumberInput value={-999} />);
+    expect(screen.getByDisplayValue('-999')).toBeInTheDocument();
+  });
+
+  test('renders decimal value', () => {
+    render(<NumberInput value={5.5} />);
+    expect(screen.getByDisplayValue('5.5')).toBeInTheDocument();
+  });
+
+  test('renders negative decimal value', () => {
+    render(<NumberInput value={-5.2} />);
+    expect(screen.getByDisplayValue('-5.2')).toBeInTheDocument();
+  });
+
+  test('does not call onChange callback when the input does not change', () => {
+    const onChangeMock = jest.fn();
+
+    render(<NumberInput onChange={onChangeMock}>5</NumberInput>);
+
+    expect(onChangeMock).not.toHaveBeenCalled();
+  });
+
+  test('calls onChange callback when input changes', () => {
+    const onChangeMock = jest.fn();
+    render(<NumberInput onChange={onChangeMock}>55</NumberInput>);
+
+    const input = screen.getByRole('spinbutton');
+    userEvent.type(input, '55');
+
+    expect(onChangeMock).toHaveBeenCalledTimes(2);
+  });
+
+  test('does not call onBlur callback when input does not lose focus', () => {
+    const onBlurMock = jest.fn();
+
+    render(<NumberInput onBlur={onBlurMock}>5</NumberInput>);
+
+    const input = screen.getByRole('spinbutton');
+    userEvent.click(input);
+
+    expect(onBlurMock).not.toHaveBeenCalled();
+  });
+
+  test('calls onBlur callback when input loses focus', () => {
+    const onBlurMock = jest.fn();
+
+    render(<NumberInput onBlur={onBlurMock}>5</NumberInput>);
+
+    const input = screen.getByRole('spinbutton');
+    userEvent.click(input);
+    userEvent.click(document.body);
+
+    expect(onBlurMock).toHaveBeenCalledTimes(1);
+  });
+
+  test('removes trailing zero', () => {
+    render(<NumberInput value={120.0} />);
+    expect(screen.getByDisplayValue('120')).toBeInTheDocument();
+  });
+
+  test('removes many trailing zeroes', () => {
+    render(<NumberInput value={120.0100} />);
+    expect(screen.getByDisplayValue('120.01')).toBeInTheDocument();
+  });
+
+  test('removes leading zero', () => {
+    render(<NumberInput value={10} onChange={() => {}} />);
+  
+    const input = screen.getByRole('spinbutton');
+    userEvent.type(input, '{arrowleft}{arrowleft}0');
+    expect(input).toHaveDisplayValue('010');
+  
+    userEvent.click(document.body)
+  
+    expect(input).toHaveDisplayValue('10');
```

Comment:
Can you add tests to verify that the leading zeros are removed on decimal and negative numbers as well?